numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4811,6 +4811,49 @@ intern_strings(void)
     return 0;
 }
 
+
+/*
+ * Initializes global constants.  At some points these need to be cleaned
```

Comment:
Agreed that it would be good to clean this up -- this setting of static globals seems better than caching it in many different places.